components/ws-manager-bridge/src/bridge.ts

```diff
@@ -252,6 +253,14 @@ export class WorkspaceManagerBridge implements Disposable {
                 return;
             }
 
+            if (instance.status.version > 0 && instance.status.version >= status.statusVersion) {
+                // We've gotten an event which is older than one we've already processed. We shouldn't process the stale one.
+                span.setTag("statusUpdate.staleEvent", true);
+                this.prometheusExporter.recordStaleStatusUpdate();
+                log.debug(ctx, "Stale status update received, skipping.");
+                return;
```

Comment:
:+1:

Reply:
This request might sound strange, but: This interface is critical, and has shown in the past to quickly break in subtle ways. :lotus_position:

Reply:
I thought about this, too - and deemed the risk ok because we're doing this kind of check for prebuilds for a while now.

But yeah, the chance of breaking stuff is high here.
Return is removed.

Reply:
/hold

Reply:
/hold cancel